Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -11,10 +11,135 @@
 #include <AMReX_MultiFab.H>
 
 
+/**
+* \brief Get the value of arr in the neighbor (i_n, j_n) on the plane with normal 'dim'.
+*        I.E. If dim==0 it return arr(i, j + i_n, k + j_n),
+*             if dim==1 it return arr(i + i_n, j, k + j_n),
+*             if dim==2 it return arr(i + i_n, j + j_n, k)
+*
+* \param[in] arr \c Array4 To be accessed
+* \param[in] i, j, k the indices of the "center" cell
+* \param[in] i_n the offset of the neighbor in the first direction
+* \param[in] j_n the offset of the neighbor in the second direction
+* \param[in] dim normal direction to the plane in consideration (0 for x, 1 for y, 2 for z)
+*/
+template <class T>
+AMREX_GPU_DEVICE
```

Comment:
Since this will be most of the time a `const` function (when non-`constexpr` arguments are passed into it), we should make sure it is inlined:
```suggestion
AMREX_GPU_DEVICE AMREX_FORCE_INLINE
```